cpp/src/strings/convert/convert_datetime.cu

```diff
@@ -280,18 +280,11 @@ struct parse_datetime {
     if (units == timestamp_units::months)
       return ((year - 1970) * 12) + (month - 1);  // months are 1-12, need to 0-base it here
     auto day = timeparts[TP_DAY];
-    // The months are shifted so that March is the starting month and February
-    // (possible leap day in it) is the last month for the linear calculation
-    year -= (month <= 2) ? 1 : 0;
-    // date cycle repeats every 400 years (era)
-    constexpr int32_t erasInDays  = 146097;
-    constexpr int32_t erasInYears = (erasInDays / 365);
-    auto era                      = (year >= 0 ? year : year - 399) / erasInYears;
-    auto yoe                      = year - era * erasInYears;
-    auto doy = month == 0 ? day : ((153 * (month + (month > 2 ? -3 : 9)) + 2) / 5 + day - 1);
-    auto doe = (yoe * 365) + (yoe / 4) - (yoe / 100) + doy;
-    int32_t days =
-      (era * erasInDays) + doe - 719468;  // 719468 = days from 0000-00-00 to 1970-03-01
+    auto ymd =  // convenient chrono class handles the leap year calculations for us
+      cuda::std::chrono::year_month_day(cuda::std::chrono::year{year},
+                                        cuda::std::chrono::month{static_cast<uint32_t>(month)},
+                                        cuda::std::chrono::day{static_cast<uint32_t>(day)});
+    int32_t days = cuda::std::chrono::sys_days(ymd).time_since_epoch().count();
```

Comment:
Introducing this code was causing crash before. reason unknown.
I suppose, that doesn't happen anymore.
following line could be replaced too.
https://github.com/rapidsai/cudf/blob/78f97898a70d328d3ccc86b9964b3f9656f6bda7/cpp%2Fsrc%2Fio%2Fcsv%2Fdatetime.cuh#L136

Reply:
I would rather open a separate PR to fix this code.